packages/consensus/src/types/utreexo.cairo

```diff
@@ -72,6 +75,75 @@ pub trait UtreexoAccumulator {
     fn delete_batch(ref self: UtreexoState, proof: @UtreexoBatchProof);
 }
 
+// https://eprint.iacr.org/2019/611.pdf Algorithm 1 AddOne
+// p18
+// To prevent such an attack, we require that the data inserted into the
+// accumulator be not just the hash of a TXO, which is controllable by the
+// attacker, but instead the concatenation of the TXO data with the block
+// hash in which the TXO is confirmed. The attacker does not know the block
+// hash before the TXO is confirmed, and it is not alterable by the attacker
+// after confirmation (without significant cost). Verifiers, when inserting into
+// the accumulator, perform this concatenation themselves after checking the
+// proof of work of the block. Inclusion proofs contain this block hash data so
+// that the leaf hash value can be correctly computed.
+fn parent_hash(left: felt252, right: felt252, _block_hash: Digest) -> felt252 {
+    let parent_data = (left, right);
+    PoseidonTrait::new().update_with(parent_data).finalize()
+}
+
+pub impl UtreexoAccumulatorImpl of UtreexoAccumulator {
+    // https://eprint.iacr.org/2019/611.pdf Algorithm 1 AddOne
+    fn add(ref self: UtreexoState, outpoint: OutPoint) {
```

Comment:
Given that we will use `UtxoSet::leaves_to_add: Array<felt252>` this method should accept outpoint hash rather than outpoint.